source/grid/grid_generator.cc

```diff
@@ -2891,6 +2891,81 @@ namespace GridGenerator
 
 
 
+  template <int dim, int spacedim>
+  void
+  subdivided_hyper_L(Triangulation<dim, spacedim> &   tria,
+                     const Point<dim> &               bottom_left,
+                     const Point<dim> &               top_right,
+                     const std::vector<unsigned int> &repetitions,
+                     const std::vector<int> &         num_cells_to_cut)
+  {
+    Assert(dim > 1, ExcNotImplemented());
+    // Check the consistency of the dimensions provided.
+    AssertDimension(repetitions.size(), dim);
+    AssertDimension(num_cells_to_cut.size(), dim);
+    for (unsigned int d = 0; d < dim; ++d)
+      {
+        Assert(std::fabs(num_cells_to_cut.at(d)) <= repetitions.at(d),
+               ExcInternalError("Attempting to cut away too many cells."));
+      }
+    // Create the domain to be cut
+    Triangulation<dim, spacedim> rectangle;
+    GridGenerator::subdivided_hyper_rectangle(rectangle,
+                                              repetitions,
+                                              bottom_left,
+                                              top_right);
+    // compute the vertex of the cut step, we will cut according to the
+    // location of the cartesian coordinates of the cell centers
+    std::vector<double> h;
+    Point<dim>          cut_step;
+    for (unsigned int d = 0; d < dim; ++d)
+      {
+        h.push_back(
+          (top_right[d] - bottom_left[d]) /
+          repetitions.at(
+            d)); // mesh spacing in each direction in cartesian coordinates
+        // left to right, bottom to top, front to back
+        if (num_cells_to_cut.at(d) >= 0)
+          {
+            // cartesian coordinates of vertex location
+            cut_step[d] =
+              h.at(d) * std::fabs(num_cells_to_cut.at(d)) + bottom_left[d];
```

Comment:
Since we already know the size of `h`, I see no reason for using `std::vector::at` instead of `std::vector::operator[]` here and below.

Reply:
Just new to C++ and wasn't sure what the common practice was, but I wasn't even being consistent with my own code and had a mix of .at(d) and [d]. Changed it all to [d], thanks for pointing this out!